sklearn/impute/_base.py

```diff
@@ -199,9 +199,9 @@ def _validate_input(self, X):
                             force_all_finite=force_all_finite, copy=self.copy)
         except ValueError as ve:
             if "could not convert" in str(ve):
-                raise ValueError("Cannot use {0} strategy with non-numeric "
-                                 "data. Received datatype :{1}."
-                                 "".format(self.strategy, X.dtype.kind))
+                new_ve = ValueError("Cannot use {} strategy with non-numeric "
+                                    "data:\n{}".format(self.strategy, str(ve)))
```

Comment:
you shouldn't need `str` with `format`. But I think you might benefit from `!r` to use repr.